calculator.rb

```diff
@@ -0,0 +1,145 @@
+# Ada - Week 01
+
+# Calculator Exercise
+# Allows user to perform simple arithmetic from the command line
+
+# Operator menu
+def print_menu
+  puts "\nMAIN MENU"
+  puts "add (+)"
+  puts "subtract (-)"
+  puts "multiply (*)"
+  puts "divide (/)"
+  puts "modulo (%)"
+  puts "exponify (^)"
+end
+
+# Get user input and validate it
+def get_num
+  user_num = gets.chomp
+
+  while user_num.to_f == 0 && (user_num != "0" && user_num != "0.0")
+    print "Sorry. Please enter a valid number: "
+    user_num = gets.chomp
```

Comment:
This is great, but it does mean the user can never intentionally enter zero. Is there another solution that would allow for that?